app/objects/c_operation.py

```diff
@@ -445,21 +448,48 @@ def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors
             facts = re.findall(BasePlanningService.re_variable, executor.test) if executor.command else []
             if not facts or all(fact in op_facts for fact in facts):
                 fact_dependency_fulfilled = True
-
-        if not agent.trusted:
-            return dict(reason='Agent untrusted', reason_id=self.Reason.UNTRUSTED.value,
+        untrusted_agent = agent.paw in self.untrusted_agents
+        associated_links = set([link.id for link in self.chain if link.paw == agent.paw
+                                and link.ability.ability_id == ability.ability_id])
+        reason_description = 'Untrusted'
+
+        if agent.platform == 'unknown':
+            if not untrusted_agent:
```

Comment:
let's move the untrusted agent check to the end - basically the agent being untrusted is the lowest priority reason an ability was skipped, since if all the other conditions are met (platform, executor, priv, facts), the only reason an agent wouldn't be assigned the ability is if it's untrusted or if the operation is still running. So you can get rid of this `if` condition throughout until the very end (the last `else`)